components/director/pkg/http/types.go

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package http
+
+import (
+	"net/http"
+)
+
+//go:generate mockery -name=HTTPRoundTripper -output=automock -outpkg=automock -case=underscore
```

Comment:
Pros of having a different repo with "util" interfaces and stuff is the fact that we can reuse the existing mocking mechanism in `system-broker` (counterfeiter) and test frame with ginkgo and gomega. If we move everything to `director` we'll have to either:
- use mockery and vanilla tests, meaning that we should rewrite all existing tests
- the mechanisms diverge until we move to counterfeiter and ginkgo everywhere

Reply:
lets keep it here, for now. Let's see in the future when we have more utils if it will make sense to have a separate component